From 97c59fbed9fecb4d5b79ea0fc678cd68cc581d59 Mon Sep 17 00:00:00 2001
From: linmin <linmin@eswincomputing.com>
Date: Mon, 28 Oct 2024 16:44:02 +0800
Subject: [PATCH 203/534] perf:change smmu print level,optimize ccache flush

Changelogs:
1.Changed the smmu event print level to dev_dbg for reducing the log
  when dsp cache prediction access to the invalid iova.
2.Move mb(); outside of the for loop of ccache_flush64_range() to reduce
  the time cost of flusing cache.

Signed-off-by: linmin <linmin@eswincomputing.com>
---
 drivers/soc/sifive/sifive_ccache.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/soc/sifive/sifive_ccache.c b/drivers/soc/sifive/sifive_ccache.c
index a88123c81ace..2010ccd64f86 100644
--- a/drivers/soc/sifive/sifive_ccache.c
+++ b/drivers/soc/sifive/sifive_ccache.c
@@ -130,8 +130,8 @@ static void ccache_flush64_range(phys_addr_t paddr, size_t size)
 	for (line = paddr; line < paddr + size;
 	line += SIFIVE_CCACHE_FLUSH64_LINE_LEN) {
 		writeq(line, ccache_base + SIFIVE_CCACHE_FLUSH64);
-		mb();
 	}
+	mb();
 }
 
 static const struct riscv_nonstd_cache_ops ccache_cmo_ops __initdata = {
-- 
2.53.0

